src/group_theory/subgroup/pointwise.lean

```diff
@@ -53,6 +53,11 @@ lemma pointwise_smul_def {a : α} (S : subgroup G) :
 lemma smul_mem_pointwise_smul (m : G) (a : α) (S : subgroup G) : m ∈ S → a • m ∈ a • S :=
 (set.smul_mem_smul_set : _ → _ ∈ a • (S : set G))
 
+lemma mem_smul_pointwise_iff_exists (m : G) (a : α) (S : subgroup G) :
+ m ∈ a • S ↔ ∃ (s : S), m = a • s :=
+⟨λ ⟨s, hs1, hs2⟩, ⟨⟨s, hs1⟩, by { rw ←hs2, refl }⟩,
```

Comment:
Is the proof just `subgroup.mem_map`?

Reply:
Alternatively, there should be a lemma in `algebra.pointwise` that makes the same statements about sets that will work as a proof.

Reply:
I find this:
```lean
lemma mem_smul_set [has_scalar α β] {t : set β} : x ∈ a • t ↔ ∃ y, y ∈ t ∧ a • y = x := iff.rfl
```

Reply:
but using `set.mem_smul_set` doesn't make proof shorter
```lean
lemma mem_smul_pointwise_iff_exists (m : G) (a : α) (S : subgroup G) :
 m ∈ a • S ↔ ∃ (s : S), m = a • s :=
begin
  erw set.mem_smul_set, split; rintros ⟨s, h⟩; use s,
  exact h.1, erw h.2,
  refine ⟨s.2, _⟩, erw h,
end
```

Reply:
I would recommend changing the statement to match `mem_smul_set`

Reply:
now that all these lemmas are just one line long, do we still really need them?

Reply:
Yes, because the lemmas are syntactically different, so rw doesn't work on the set one here. Try proving this with only `rw` and you'll have a hard time.

This entire file contains these one-liner proofs.